### ISSUE TITLE
SystemProperty Test Helpers

### ISSUE THREAD

**Issue by sksamuel** [kotlintest/kotlintest#524]
Might be useful to have a trait that offers some functionality to perform a test with a system property enabled, and then restores the system property to whatever it was before.

```
class MyTest : FunSpec({
  test("foo") {
     withSysProperty("key", "value") {
       // test here
     }
  }
```

**Comment by sksamuel** [comment]
Your thoughts @Kerooker

**Comment by Kerooker** [comment]
Find it great.
Would be good to have a plural form too `withSysProperties`

**Comment by LeoColman** [comment]
Closed as per #608
